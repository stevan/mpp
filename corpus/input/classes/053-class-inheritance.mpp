# Base class
class Animal {
    field $name;

        return "Some sound";

# Single inheritance with method without parentheses
class Dog :isa(Animal) {
        return "Woof!";

# Inheritance with qualified class name and method with parentheses
class Cat :isa(My::Animal) {
        return "Meow!";

# Class with inheritance, fields, and methods
class Bird :isa(Animal) {
    field $can_fly;

        return $can_fly;

        return "Tweet!";
