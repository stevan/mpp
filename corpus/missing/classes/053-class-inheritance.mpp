    method speak {
    }
}
    method speak {
    }
}
    method speak () {
    }
}
    method fly {
    }
    method chirp () {
    }
}