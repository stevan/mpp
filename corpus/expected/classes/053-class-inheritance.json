[
  {
    "type": "Class",
    "name": "Animal",
    "body": [
      {
        "type": "Field",
        "variable": {
          "type": "Variable",
          "name": "$name"
        }
      },
      {
        "type": "Return",
        "value": {
          "type": "String",
          "value": "\"Some sound\""
        }
      }
    ]
  }
]